doc/md/canister-maintenance/cycles.md

```diff
@@ -0,0 +1,111 @@
+---
+sidebar_position: 1
+---
+
+
+# Cycles
+
+## Overview
+
+Usage of a canister's resources on ICP is measured and paid for in [cycles](/docs/current/developer-docs/getting-started/cycles/converting_icp_tokens_into_cycles).
+
+In Motoko programs deployed on ICP, each actor represents a canister and has an associated balance of cycles. The ownership of cycles can be transferred between actors. Cycles are selectively sent and received through shared function calls. A caller can choose to transfer cycles with a call, and a callee can choose to accept cycles that are made available by the caller. Unless explicitly instructed, no cycles are transferred by callers or accepted by callees.
+
+Callees can accept all, some, or none of the available cycles up to limit determined by their actor’s current balance. Any remaining cycles are refunded to the caller. If a call traps, all its accompanying cycles are automatically refunded to the caller without loss.
+
+In future, we may see Motoko adopt dedicated syntax and types to support safer programming with cycles. For now, we provide a temporary way to manage cycles through a low-level imperative API provided by the [ExperimentalCycles](../base/ExperimentalCycles.md) library in package `base`.
+
+:::note
+
+This library is subject to change and likely to be replaced by more high-level support for cycles in later versions of Motoko.
+
+:::
+
+## The `ExperimentalCycles` Library
+
+The `ExperimentalCycles` library provides imperative operations for observing an actor’s current balance of cycles, transferring cycles and observing refunds.
+
+The library provides the following operations:
+
+``` motoko no-repl
```

Comment:
Will these long one line comments render nicel? If not, I'd go back to just showing the function signatures, and then using bullets underneath to explain each one.

Reply:
addressed in https://github.com/dfinity/motoko/pull/4512